ast/compile.go

```diff
@@ -883,6 +883,47 @@ func compileSchema(goSchema interface{}) (*gojsonschema.Schema, error) {
 	return schemasCompiled, nil
 }
 
+func mergeSchemas(schemas []*gojsonschema.SubSchema) (*gojsonschema.SubSchema, error) {
+	if len(schemas) == 0 {
+		return nil, nil
+	}
+	var result = schemas[0]
+
+	for i := 0; i < len(schemas); i++ {
+		if len(schemas[i].PropertiesChildren) > 0 {
+			if !schemas[i].Types.Contains("object") {
+				if err := schemas[i].Types.Add("object"); err != nil {
+					return nil, fmt.Errorf("unable to set the type in schemas")
+				}
+			}
+		} else if len(schemas[i].ItemsChildren) > 0 {
+			if !schemas[i].Types.Contains("array") {
+				if err := schemas[i].Types.Add("array"); err != nil {
+					return nil, fmt.Errorf("unable to set the type in schemas")
+				}
+			}
+		}
+	}
+
+	for i := 1; i < len(schemas); i++ {
+		if result.Types.String() != schemas[i].Types.String() {
+			return nil, fmt.Errorf("unable to merge these schemas")
```

Comment:
Maybe we can add a brief "type mismatch" info here, with the two types that clashed? I might be making things up, but I can imagine how this could be difficult to debug if encountered by users.